src/UserService.Business/Commands/User/EditUserCommand.cs

```diff
@@ -22,337 +19,327 @@
+                userId,
+                _httpContextAccessor.HttpContext.GetUserId())).Result;
+
+        if (!response.Message.IsSuccess || !response.Message.Body)
         {
-            Guid? avatarImageId = null;
-
-            if (avatarRequest == null)
-            {
-                return null;
-            }
-
-            Guid userId = _httpContextAccessor.HttpContext.GetUserId();
-
-            string errorMessage = $"Can not add avatar image to user with id {userId}. Please try again later.";
-
-            try
-            {
-                var imageRequest = IAddImageRequest.CreateObj(
-                    avatarRequest.Name,
-                    avatarRequest.Content,
-                    avatarRequest.Extension,
-                    userId);
-                var response = _rcImage.GetResponse<IOperationResult<Guid>>(imageRequest).Result;
-
-                if (!response.Message.IsSuccess)
-                {
-                    _logger.LogWarning(
-                        "Can not add avatar image to user with id {UserId}. Reason: '{Errors}'",
-                        userId,
-                        string.Join(',', response.Message.Errors));
-
-                    errors.Add(errorMessage);
-                }
-                else
-                {
-                    avatarImageId = response.Message.Body;
-                }
-            }
-            catch (Exception exc)
-            {
-                _logger.LogError(exc, "Can not add avatar image to user with id {UserId}", userId);
-
-                errors.Add(errorMessage);
-            }
-
-            return avatarImageId;
+          const string warningMessage = logMessage + "Errors: {Errors}";
+          _logger.LogWarning(warningMessage, roleId, userId, string.Join("\n", response.Message.Errors));
+
+          errors.Add(errorMessage);
         }
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage, roleId, userId);
+
+        errors.Add(errorMessage);
+      }
+    }
+
+    private Guid? GetAvatarImageId(AddImageRequest avatarRequest, List<string> errors)
+    {
+      Guid? avatarImageId = null;
 
-        #endregion
-
-        public EditUserCommand(
-            IUserRepository userRepository,
-            IUserCredentialsRepository credentialsRepository,
-            IPatchDbUserMapper mapperUser,
-            IAccessValidator accessValidator,
-            ILogger<EditUserCommand> logger,
-            IRequestClient<IAddImageRequest> rcImage,
-            IRequestClient<IChangeUserDepartmentRequest> rcDepartment,
-            IRequestClient<IChangeUserPositionRequest> rcPosition,
-            IRequestClient<IChangeUserRoleRequest> rcRole,
-            IRequestClient<IChangeUserOfficeRequest> rcOffice,
-            IHttpContextAccessor httpContextAccessor,
-            IBus bus)
+      if (avatarRequest == null)
+      {
+        return null;
+      }
+
+      Guid userId = _httpContextAccessor.HttpContext.GetUserId();
```

Comment:
maybe it is better if we create global variable with requester user Id? we use it in all private methods and every time pull it.

Reply:
I don't think so, it is not property of class in meaning of logic, i don't know how to say better

Reply:
also i think it is unsafety